examples/dev_example/docs/build.jl

```diff
@@ -0,0 +1,28 @@
+import Lexicon.Elements: config, section, document, page
+
+import Lexicon.Render: save, markdown
+
+import Docile: Cache, Collector
+
+out = document("TestDocumentation",
+    section("Introduction",
+        page("index", "index.md"),
+        ),
+    section("Docile-API",
+        page("Cache", config(style_obj="###"),
+            "# Cache",
+            "## Macros",
+            "Some intermediate text describing more about macros.",
+            (Cache, config(objfilter=[:macro])),
+            "## Methods",
+            (Cache, config(objfilter=[:method])),
+            ),
+        page("Collector", config(objfilter=[:method, :macro], style_obj="###"),
+            "# Collector",
+            "## Methods & Macros",
+            Collector,
+            ),
+        ),
+    )
```

Comment:
> Given we're storing a parent reference it's easy to check whether a section is nested inside a page I think and handle accordingly.

true - myself I would like to keep the section separate: example for high level autogenerated package section something like

```
section(:API, 
          [pkg or other multi-modules]
)
```

should aut-generate for each module a separate page - which would make using Lexicon for smaller projects very easy.

> Could we use a Symbol for the name then?

I guess - but it would pu some limitations on what is a valid name - which generates also `folders for makedown output` and should later generate the page structure for `mkdocs.yml`

So one can not anymore do something like:
```
section("My Section Output Folder",

)
```

> On the other hand: Potentially we could rename objs to either objects or entries or docstrings. I think that would be a better naming if we can't use section instead. How's that sound?

given the 3 options this would be my prefered list. the first 2 are fine the 3rd I would like to avoid.

* tuple when using a config: I think using the tuple does not intorduce more new features than the next option. Also if we have good defaults: for many smaller projects there might be no need to set at this level a config.
* docstrings()
* nested section in pages